### ISSUE TITLE
we should have a way of undoing changed that break the schema

### ISSUE THREAD

**Issue by glommer** [chiselstrike/chiselstrike#1610]
It doesn't matter how careful we are, there will always be changes that end up preventing chiseld from start.
One recent example is #1605, that passed the apply cycle tests and managed to commit the schema, but would then fail to load the schema on startup.

Another example where we may find ourselves in problematic situation is if we ever break compatibility in a future version (which can happen by accident). If chiseld can't boot, we can't apply, and if we can't apply, there's nothing we can do to fix this.

There has to be a way to either apply-on-start (which is hard, because the apply logic lives on the client), or start on wait mode, waiting for an apply. This would allow us to, in the worst case, apply schema changes to get chiseld back in shape.

**Comment by honzasp** [comment]
Yes, I think that this is one reason for performing the database work in `chisel apply` rather than in `chiseld`: if the apply gets broken and `chiseld` does not start, you just need to fix `chisel apply` and run it again, because no interaction with `chiseld` is necessary.
